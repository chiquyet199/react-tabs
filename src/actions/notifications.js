import {
  INFO_NOTIFICATION,
  ERROR_NOTIFICATION,
  WARNING_NOTIFICATION,
  CLEAR_NOTIFICATION,
} from "./types";
import {sleep} from "../utils/common";
import {loggedOut} from "./auth";

export const clearNotification = {
  type: CLEAR_NOTIFICATION,
};

export const expireNotification = async dispatch => {
  await sleep(3000);

  dispatch(clearNotification);
};

export const infoNotification = message => dispatch => {
  dispatch({
    type: INFO_NOTIFICATION,
    message,
  });

  expireNotification(dispatch);
};

export const errorNotification = error => dispatch => {
  if (error && error.status === 401) {
    dispatch(loggedOut());
  } else {
    dispatch({
      type: ERROR_NOTIFICATION,
      message: error.message,
    });
    expireNotification(dispatch);
  }
};

export const warningNotification = message => dispatch => {
  dispatch({
    type: WARNING_NOTIFICATION,
    message,
  });

  expireNotification(dispatch);
};
