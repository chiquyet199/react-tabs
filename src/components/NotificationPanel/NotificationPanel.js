import styled from "styled-components/native";
import {isWeb} from "../../utils/common";

export const Container = styled.View`
  position: ${({topbar}) =>
    topbar === "true" && isWeb ? "absolute" : "relative"};
  left: 0;
  top: ${({topbar}) => (topbar === "true" && isWeb ? "100%" : "0")};
  background-color: ${({color, theme}) => color || theme.colors.none};
  opacity: 0.8;
  min-height: 38px;
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding-horizontal: 15px;
  margin-top: ${({topmargin}) =>
    topmargin === 0 || topmargin ? `${topmargin}px` : "-15px"};
  margin-bottom: ${({bottommargin}) =>
    bottommargin === 0 || bottommargin ? `${bottommargin}px` : "0"};
`;

export const NotificationText = styled.Text`
  padding: 0;
  color: white;
  flex-grow: 3;
  color: ${({color, theme}) => color || theme.colors.lightSteel};
`;

export const TextWrapper = styled.View`
  flex-grow: 2;
  padding-left: 15px;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  flex-wrap: nowrap;
  padding: 10px;
`;
