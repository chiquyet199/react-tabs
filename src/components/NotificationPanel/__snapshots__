/Users/edward/Desktop/canopy-frontend/packages/canopy-frontend-common/src/components/NotificationPanel/__native_snapshots__