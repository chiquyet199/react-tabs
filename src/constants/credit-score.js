// THIS CODE IS COPIED FROM OLD "canopy-legacy" REPO
// Definition of common errors.
const NO_TRACE = "NO_TRACE";
const NO_TRACE_VALUE = -998;
const UNDEFINED_SCORE = "UNDEFINED_SCORE";
export const SCORE_MAX_VALUE = 999;
// Definition for the scores.
// Based on https://www.experian.co.uk/consumer/experian-credit-score.html
export const SCORES = {
  VERY_POOR: "VERY_POOR",
  POOR: "POOR",
  FAIR: "FAIR",
  GOOD: "GOOD",
  EXCELLENT: "EXCELLENT",
};

// Definition of the scores level within their ranges.
export const SCORE_RANGES = {
  [SCORES.VERY_POOR]: {from: 0, to: 560},
  [SCORES.POOR]: {from: 561, to: 720},
  [SCORES.FAIR]: {from: 721, to: 880},
  [SCORES.GOOD]: {from: 881, to: 960},
  [SCORES.EXCELLENT]: {from: 961, to: SCORE_MAX_VALUE},
};

// Values that are consider trusty.
const TRUSTY_SCORE_RANGE = [
  SCORES.POOR,
  SCORES.FAIR,
  SCORES.GOOD,
  SCORES.EXCELLENT,
];

/**
 * Get the score range based on the score literal.
 * @param  {String} score
 * @return {Object}
 */
export const getScoreRanges = score => SCORE_RANGES[score];

/**
 * A score shouldn't go beyond 999, which is the maximum at 'excellent' score level.
 * If the score goes beyond 999, just return 999.
 * @param  {Number} score
 * @return {Number}
 */
const normalizeScore = score =>
  Math.min(score, getScoreRanges(SCORES.EXCELLENT).to);

/**
 * Helper function to check if a score appears under a certain category.
 * @param  {Number} score
 * @param  {Object} range
 * @return {Boolean}
 */
const scoreIsInRange = (score, range) =>
  score >= range.from && score <= range.to;

/**
 * Check if the score is valued as 'no trace'.
 * @param  {Number}  score
 * @return {Boolean}
 */
const isNoTrace = score => score === NO_TRACE_VALUE;

/**
 * Check if the score is cataloged as 'very poor'.
 * @param  {Number}  score
 * @return {Boolean}
 */
const isVeryPoor = score =>
  scoreIsInRange(score, getScoreRanges(SCORES.VERY_POOR));

/**
 * Check if the score is cataloged as 'poor'.
 * @param  {Number}  score
 * @return {Boolean}
 */
const isPoor = score => scoreIsInRange(score, getScoreRanges(SCORES.POOR));

/**
 * Check if the score is cataloged as 'fair'.
 * @param  {Number}  score
 * @return {Boolean}
 */
const isFair = score => scoreIsInRange(score, getScoreRanges(SCORES.FAIR));

/**
 * Check if the score is cataloged as 'good'.
 * @param  {Number}  score
 * @return {Boolean}
 */
const isGood = score => scoreIsInRange(score, getScoreRanges(SCORES.GOOD));

/**
 * Check if the score is cataloged as 'excellent'
 * @param  {Number}  score
 * @return {Boolean}
 */
const isExcellent = score =>
  scoreIsInRange(score, getScoreRanges(SCORES.EXCELLENT));

/**
 * Get the score level (like 'VERY_POOR', 'POOR', etc.) for the score.
 * It will also check for possible errors like 'NO_TRACE'.
 *
 * @param  {Number} score
 * @return {String}
 */
export const getScoreLevel = score => {
  // Normalize the score, making it go as much as 999.
  const normalizedScore = normalizeScore(score);

  if (isNoTrace(normalizeScore)) {
    return NO_TRACE;
  }

  if (isVeryPoor(normalizedScore)) {
    return SCORES.VERY_POOR;
  }

  if (isPoor(normalizedScore)) {
    return SCORES.POOR;
  }

  if (isFair(normalizedScore)) {
    return SCORES.FAIR;
  }

  if (isGood(normalizedScore)) {
    return SCORES.GOOD;
  }

  if (isExcellent(normalizedScore)) {
    return SCORES.EXCELLENT;
  }

  return UNDEFINED_SCORE;
};

/**
 * Check if the score is consider as trusty.
 * @param  {Number}  score
 * @return {Boolean}
 */
export const isTrustyScore = score =>
  TRUSTY_SCORE_RANGE.includes(getScoreLevel(score));
