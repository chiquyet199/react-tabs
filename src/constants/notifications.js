export const NOTIFICATION_TYPES = Object.freeze([
  "error",
  "warning",
  "info",
  "rent_passport_info",
  "rent_passport_warning",
  "rent_passport_error",
]);

export const NOTIFICATION_ICON_COLORS = theme => ({
  error: {
    bg: theme.colors.alertRed,
    text: theme.colors.white,
    iconColor: theme.colors.white,
  },
  info: {
    bg: theme.colors.successGreen,
    text: theme.colors.white,
    iconColor: theme.colors.white,
  },
  warning: {
    bg: theme.colors.warning,
    text: theme.colors.white,
    iconColor: theme.colors.white,
  },
  rent_passport_info: {
    bg: theme.colors.none,
    text: theme.colors.successGreen,
    iconColor: theme.colors.successGreen,
  },
  rent_passport_warning: {
    bg: theme.colors.rentPassportWarningBg,
    text: theme.colors.orange,
    iconColor: theme.colors.orange,
  },
  rent_passport_error: {
    bg: theme.colors.rentPassportErrorBg,
    text: theme.colors.alertRed,
    iconColor: theme.colors.alertRed,
  },
});
