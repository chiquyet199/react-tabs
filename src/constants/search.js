export const WAIT_INTERVAL = 1000;
export const ENTER_KEY = 13;
