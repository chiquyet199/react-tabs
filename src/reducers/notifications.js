import {
  INFO_NOTIFICATION,
  ERROR_NOTIFICATION,
  WARNING_NOTIFICATION,
  CLEAR_NOTIFICATION,
} from "../actions/types";

const initialState = {
  type: null,
  message: "",
};

export default (state = initialState, {type, message}) => {
  switch (type) {
    case ERROR_NOTIFICATION:
      return {
        type: "error",
        icon: "error",
        message: message || "",
      };
    case INFO_NOTIFICATION:
      return {
        type: "info",
        icon: "tick",
        message,
      };
    case WARNING_NOTIFICATION:
      return {
        type: "warning",
        icon: "error",
        message,
      };
    case CLEAR_NOTIFICATION:
      return initialState;
    default:
      return state;
  }
};
