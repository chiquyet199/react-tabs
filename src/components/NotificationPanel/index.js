import React from "react";
import {func, node, number, bool, string, oneOf} from "prop-types";
import {themed} from "../../utils/common";
import Icon from "../Icon";
import {Container, NotificationText, TextWrapper} from "./NotificationPanel";
import {
  NOTIFICATION_TYPES,
  NOTIFICATION_ICON_COLORS,
} from "../../constants/notifications";

export const NotificationPanel = ({
  text,
  content,
  textColor,
  iconColor,
  iconName,
  handlePress,
  type,
  theme,
  topmargin,
  bottommargin,
  isTopbar,
}) => {
  const colors = NOTIFICATION_ICON_COLORS(theme)[type];

  if (text || content) {
    return (
      <Container
        theme={theme}
        color={colors.bg}
        bottommargin={bottommargin}
        topmargin={topmargin}
        topbar={isTopbar && isTopbar.toString()}
      >
        {text && (
          <TextWrapper>
            <Icon
              name={iconName}
              size={26}
              color={iconColor || colors.iconColor}
            />
            <NotificationText
              theme={theme}
              color={textColor || iconColor || colors.text}
            >
              {text}
            </NotificationText>
            {handlePress && (
              <Icon
                name={iconName}
                size={26}
                color={colors.iconColor}
                onPress={handlePress}
                onClick={handlePress}
              />
            )}
          </TextWrapper>
        )}
        {content && content}
      </Container>
    );
  }

  return null;
};

NotificationPanel.defaultProps = {
  type: "",
};
NotificationPanel.propTypes = {
  handlePress: func,
  text: string,
  content: node,
  textColor: string,
  iconColor: string,
  iconName: string,
  type: oneOf(NOTIFICATION_TYPES),
  topmargin: number,
  bottommargin: number,
  isTopbar: bool,
};

export default themed(NotificationPanel);
