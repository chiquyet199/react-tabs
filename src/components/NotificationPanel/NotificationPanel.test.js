import React from "react";
import {Platform} from "react-native";
import {shallow} from "enzyme";
import {NOTIFICATION_ICON_COLORS} from "../../constants/notifications";
import {NotificationPanel} from "./index";
import {Container, NotificationText} from "./NotificationPanel";
import {extend} from "underscore";
import Icon from "../../components/Icon";

describe("Notification Panel component", () => {
  it("render properly - " + Platform.OS, () => {
    expect(getComponent()).toMatchSnapshot();
  });

  it("Sets proper color props depending on error types", () => {
    const {theme} = getMockProps();
    const colorProps = NOTIFICATION_ICON_COLORS(theme);
    const types = [
      "error",
      "warning",
      "info",
      "rent_passport_info",
      "rent_passport_warning",
      "rent_passport_error",
    ];
    types.forEach(async type => {
      const comp = await getComponent({
        type,
        iconColor: undefined,
        textColor: undefined,
      });
      expect(comp.find(Container).prop("color")).toBe(colorProps[type].bg);
      expect(comp.find(Icon).prop("color")).toBe(colorProps[type].iconColor);
      expect(comp.find(NotificationText).prop("color")).toBe(
        colorProps[type].text,
      );
    });
  });

  it("Renders content if passed", async () => {
    const content = <NotificationText>Text</NotificationText>;
    const comp = await getComponent({
      content,
      text: undefined,
    });
    expect(comp.find(NotificationText)).toExist();
  });

  it("Renders action Icon if handlePress is passed", () => {
    const handlePress = jest.fn();
    const comp = getComponent({
      handlePress,
    });
    expect(comp.find(Icon)).toHaveLength(2);
  });

  const getComponent = props => {
    const parsedProps = extend(getMockProps(), props);

    return shallow(<NotificationPanel {...parsedProps} />);
  };

  const getMockProps = () => ({
    handlePress: () => {},
    text: "Notification text",
    content: undefined,
    textColor: "#ff0000",
    iconColor: "#ff0000",
    iconName: "warning",
    type: "error",
    topmargin: 5,
    bottommargin: 5,
    isTopbar: false,
    theme: {
      colors: {
        lightSteel: "#e3eaf2",
        alertRed: "#f16651",
        warning: "#fffcf3",
        orange: "#f1bf51",
        white: "#ffffff",
        successGreen: "#5dc471",
        rentPassportErrorBg: "#fff5f5",
        rentPassportWarningBg: "#fffcf3",
      },
    },
  });
});
