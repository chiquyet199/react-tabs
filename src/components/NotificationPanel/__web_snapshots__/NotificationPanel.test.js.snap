// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Panel component render properly - web 1`] = `
<Styled(View)
  bottommargin={5}
  color="#f16651"
  theme={
    Object {
      "colors": Object {
        "alertRed": "#f16651",
        "lightSteel": "#e3eaf2",
        "orange": "#f1bf51",
        "rentPassportErrorBg": "#fff5f5",
        "rentPassportWarningBg": "#fffcf3",
        "successGreen": "#5dc471",
        "warning": "#fffcf3",
        "white": "#ffffff",
      },
    }
  }
  topbar={false}
  topmargin={5}
>
  <Styled(View)>
    <Icon
      allowFontScaling={false}
      color="#ff0000"
      name="warning"
      size={26}
    />
    <Styled(Text)
      color="#ff0000"
      theme={
        Object {
          "colors": Object {
            "alertRed": "#f16651",
            "lightSteel": "#e3eaf2",
            "orange": "#f1bf51",
            "rentPassportErrorBg": "#fff5f5",
            "rentPassportWarningBg": "#fffcf3",
            "successGreen": "#5dc471",
            "warning": "#fffcf3",
            "white": "#ffffff",
          },
        }
      }
    >
      Notification text
    </Styled(Text)>
    <Icon
      allowFontScaling={false}
      color="#ffffff"
      name="warning"
      onClick={[Function]}
      onPress={[Function]}
      size={26}
    />
  </Styled(View)>
</Styled(View)>
`;
